### ISSUE TITLE
Spike: Monitoring/Logging when our own backend submission fails

### ISSUE THREAD

**Issue by aplatt-coforma** [department-of-veterans-affairs/va.gov-team#70487]
### Summary

- What issues could potentially have gone wrong when trying to create a PDF for pensions?

- What creates the PDF?  Is it **EVSS/getPDF**. If not, what was it a service in vets-api (modules/appeals_api/app/services/appeals_api/pdf_construction/generator.rb) that generates the PDF and then uploads it to central mail?

- What is your best guess as to how error prone it was based upon what you know?

### Notes:
- Sharepoint doc: [https://dvagov.sharepoint.com/❌/s/vaabdvro/EZt9wsv0OpdLgH6595eNdmUBVNaLLe2FAPTSZisiHpB1Pw?e=TYkSHm](https://dvagov.sharepoint.com/:x:/s/vaabdvro/EZt9wsv0OpdLgH6595eNdmUBVNaLLe2FAPTSZisiHpB1Pw?e=TYkSHm)
- Slack thread: https://dsva.slack.com/archives/C057W3N9K2S/p1700677329672019

**Comment by mjknight50** [comment]
@aplatt-coforma  Does this ticket already cover what you are asking for?
https://github.com/department-of-veterans-affairs/va.gov-team/issues/70462

**Comment by aplatt-coforma** [comment]
@mjknight50 - good find. I can close this as a duplicate of that ticket.

**Comment by aplatt-coforma** [comment]
@mjknight50 - Actually, I think this ticket was more to answer Emily's question:  "Do we have any error monitor when our own backend errors-out? Was there monitoring in the past?" And to help fill out some of the missing pieces in this spreadsheet: https://dvagov.sharepoint.com/:x:/s/vaabdvro/EZt9wsv0OpdLgH6595eNdmUBVNaLLe2FAPTSZisiHpB1Pw?e=TYkSHm (Sharepoint)

The ticket you reference is more of an implementation ticket right?

**Comment by danlim715** [comment]
@emilytheis @aplatt-coforma @mjknight50 @vbellissimo 

If the form data doesn't match the json schema definition, then we do throw a validation error and it is logged to datadog, but this happens at the SavedClaim creation point in the logic, which is before the PDF generation logic is kicked off.

In the claim submission logic (which includes the pdf mapping logic), it does appear that we have some logging for success and failure attempts (to raven-sentry, but not Rails.logger). And now we retry submissions for up to 14 times with Datadog monitoring and additional logging if all attempts have been exhausted.

In terms of potential failure points in the code, I did discover a small bug in the pdf logic during my testing that unveiled a possible `'sort_by': comparison of Array with Array failed (ArgumentError)` in certain scenarios. I've added the fix in [this](https://github.com/department-of-veterans-affairs/vets-api/pull/14702) PR. As we add more logic to map fields, it's very possible that we introduce some new points of failure while trying to parse or extract certain data if proper null checks, type checks, etc. aren't done. We will need to be careful and err on the side of defensive programming to minimize these occurrences.
